packages/@aws-cdk/aws-ecs-patterns/lib/base/queue-processing-service-base.ts

```diff
@@ -273,8 +287,20 @@ export abstract class QueueProcessingServiceBase extends CoreConstruct {
     this.secrets = props.secrets;
 
     // Determine the desired task count (minimum) and maximum scaling capacity
-    this.desiredCount = props.desiredTaskCount ?? 1;
-    this.maxCapacity = props.maxScalingCapacity || (2 * this.desiredCount);
+    if (!this.node.tryGetContext(cxapi.ECS_REMOVE_DEFAULT_DESIRED_COUNT)) {
+      this.desiredCount = props.desiredTaskCount != null ? props.desiredTaskCount : 1;
```

Comment:
```suggestion
      this.desiredCount = props.desiredTaskCount ?? 1;
```

Reply:
Sadly the reason I had to change this was for some reason the unit test was failing when desiredTaskCount was set to 0, it was being overridden to 1. That's why I added the explicit conditions.

Reply:
I tried this refactor locally and the tests pass. Using [null coalescing](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-7.html#nullish-coalescing) is a more concise equivalent to what you have.

Reply:
Updated.